src/main/java/cubicchunks/worldgen/GeneratorPipeline.java

```diff
@@ -122,7 +122,7 @@ public void generate(Cube cube) {
 	public void generate(Cube cube, GeneratorStage targetStage) {
 
 		// Make sure the proper target stage is set.
-		if (cube.getTargetStage().precedes(targetStage)) {
+		if (cube.getTargetStage() == null || cube.getTargetStage().precedes(targetStage)) {
```

Comment:
Currently yes, because it is no longer set in all paths of ServerCubeCache.loadCube, but I'll change that to get rid of that additional check. Cleaner without it.